src/test/java/com/codeus/winter/config/DefaultBeanFactoryTest.java

```diff
@@ -276,4 +285,40 @@ void testCreateBeanShouldThrowExceptionWhenBeanIsNotUnique() {
                 exception.getMessage()
         );
     }
+
+    @Test
+    @DisplayName("Should apply bean post processors before initialization")
+    void testApplyBeanPostProcessorsBeforeInitialization() {
+        DefaultBeanFactory beanFactory = new DefaultBeanFactory(Map.of(
+                "BeanA", beanDefinitionA
+        ));
+        BeanPostProcessor beanPostProcessor = spy(BeanPostProcessor.class);
+        BeanA beanAfterPostProcessing = new BeanA();
+        when(beanPostProcessor.postProcessBeforeInitialization(any(), anyString())).thenReturn(beanAfterPostProcessing);
+        beanFactory.addBeanPostProcessor(beanPostProcessor);
+        beanFactory.initializeBeans();
+
+        verify(beanPostProcessor, times(1)).postProcessBeforeInitialization(any(), anyString());
+        BeanA beanA = beanFactory.getBean(BeanA.class);
+        assertNotNull(beanA);
+        assertEquals(beanAfterPostProcessing, beanA);
+    }
+
+    @Test
+    @DisplayName("Should apply bean post processors after initialization")
+    void testApplyBeanPostProcessorsAfterInitialization() {
+        DefaultBeanFactory beanFactory = new DefaultBeanFactory(Map.of(
+                "BeanA", beanDefinitionA
+        ));
+        BeanPostProcessor beanPostProcessor = spy(BeanPostProcessor.class);
+        BeanA beanAfterPostProcessing = new BeanA();
+        when(beanPostProcessor.postProcessAfterInitialization(any(), anyString())).thenReturn(beanAfterPostProcessing);
+        beanFactory.addBeanPostProcessor(beanPostProcessor);
+        beanFactory.initializeBeans();
+
+        verify(beanPostProcessor, times(1)).postProcessAfterInitialization(any(), anyString());
```

Comment:
`times(1)` part can be removed, it is implicitly called in the `T verify(T mock)`. 

Please do the same for the tests above.